sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -36,18 +36,66 @@ namespace sycl {
 
 // Forward declarations
 class queue;
-namespace detail {
-class queue_impl;
-} // namespace detail
 
 namespace detail {
 
+// Used to represent a type of an extended member
+enum class ExtendedMembersType: unsigned int {
+  HANDLER_KERNEL_BUNDLE = 0,
+};
+
+// Holds a pointer to an object of an arbitrary type and an ID value which
+// should be used to understand what type pointer points to.
+// Used as to extend handler class without introducing new class members which
+// would change handler layout.
+struct ExtendedMemberT {
+  ExtendedMembersType MType;
+  std::shared_ptr<void> MData;
+};
+
+static std::shared_ptr<std::vector<ExtendedMemberT>>
```

Comment:
NIT: why not simple pointer? I wonder if the result of convertToExtendedMembers can escape and get referenced by user structures?

Reply:
Hypothetically it can. It seems logical to not provide interfaces using which naively one can get very hard to debug problems .